### PR TITLE
Create config file for Mockery instead of using explicit CLI flags in the Makefile

### PR DESCRIPTION
**Pull request by jesslourenco**
## Which problem is this PR solving?
- Part of #5569

## Description of the changes
- Created the configuration file (.mockery.yaml) to match what was being done by `make generate-mocks`
- Removed a few mock files from es/mocks that were defined within the subpackage `client`. These files are still (auto) generated but within es/client/mocks only

## How was this change tested?
- manually by running `mockery`
- by running `make test`

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
please make sure all commits are signed

File: .mockery.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  github.com/jaegertracing/jaeger/pkg/es:
+    config:
+      all: true
+      mockname: "{{.InterfaceName}}"
+      filename: "{{.InterfaceName}}.go"
+  github.com/jaegertracing/jaeger/pkg/es/client:
+    interfaces:
+      IndexAPI:
+        config:
+          filename: "index_client.go"
```
Comment (yurishkuro) [inline .mockery.yaml:16]: there seems to be a lot more details (or customizations) here compared to what we had in the makefile, are they all required? Can we rely more on the defaults?

also, as part of the PR we need to change the target in the makefile, so that CI can validate these changes

File: .mockery.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+with-expecter: false
+disable-version-string: true
+outpkg: "mocks"
+dir: "{{.InterfaceDir}}/mocks/"
+filename: "{{ .InterfaceName | snakecase }}.go"
```
Comment (yurishkuro) [inline .mockery.yaml:5]: do we need snakecase? I see most of our mocks just follow the interface name. If we have a single package with snakecase, we can just update it.

File: .mockery.yaml (hunk 1 above)
Comment (jesslourenco) [reply .mockery.yaml:16]: sure! I can clean up all except the details for `github.com/jaegertracing/jaeger/pkg/es/client` because these files don't match the rest (eg. mocks start with a "Mock" prefix).

Comment (jesslourenco) [reply .mockery.yaml:16]: about CI, are there any further tests you recommend running locally aside from `make test-ci`?

Comment (yurishkuro) [reply .mockery.yaml:16]: why is `es/client` special? I prefer everything to be uniform, if this package diverges in conventions from the others, we should just fix that.

Comment (jesslourenco) [reply .mockery.yaml]: I have no idea; it seems to follow mockery conventions, though. I will update the files

File: .mockery.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  github.com/jaegertracing/jaeger/storage/spanstore: 
+    config:
+      all: true
+  github.com/jaegertracing/jaeger/proto-gen/storage_v1: 
+    config:
+      all: true
+  github.com/jaegertracing/jaeger/pkg/es/client:
+    config:
+      all: true
+
```
Comment (yurishkuro) [inline .mockery.yaml:20]: We seem to have a lot of other packages for which we probably just generated mocks manually once. Would be nice to include them in this config too (could be a separate PR):
```
$ find . -name mocks
./crossdock/services/mocks
./cmd/ingester/app/processor/mocks
./cmd/ingester/app/consumer/mocks
./cmd/collector/app/sanitizer/cache/mocks
./proto-gen/storage_v1/mocks
./plugin/sampling/leaderelection/mocks
./plugin/storage/grpc/mocks
./storage/mocks
./storage/metricsstore/mocks
./storage/dependencystore/mocks
./storage/spanstore/mocks
./storage/samplingstore/mocks
./pkg/kafka/mocks
./pkg/cassandra/mocks
./pkg/es/mocks
./pkg/es/client/mocks
./pkg/distributedlock/mocks
```

**Review by yurishkuro — APPROVED** [review]

File: .mockery.yaml (hunk 3 above)
Comment (yurishkuro) [reply .mockery.yaml:20]: I'll add it back to the ticket

**Comment by yurishkuro** [comment]
Thanks!
